client/src/main/java/io/pravega/client/batch/impl/SegmentIteratorImpl.java

```diff
@@ -22,35 +22,48 @@
 import io.pravega.client.segment.impl.Segment;
 import io.pravega.client.segment.impl.SegmentInputStreamFactory;
 import io.pravega.client.segment.impl.SegmentTruncatedException;
-import io.pravega.client.stream.Serializer;
-import io.pravega.client.stream.TruncatedDataException;
+import io.pravega.client.stream.*;
 
+import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.util.NoSuchElementException;
 import java.util.concurrent.TimeoutException;
-
+import io.pravega.common.ObjectBuilder;
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import io.pravega.common.util.ByteArraySegment;
 import io.pravega.common.util.Retry;
+import lombok.Builder;
 import lombok.Getter;
+import lombok.SneakyThrows;
 import lombok.extern.slf4j.Slf4j;
 
 @Beta
 @Slf4j
 public class SegmentIteratorImpl<T> implements SegmentIterator<T> {
-
+    @Getter
     private final Segment segment;
     private final Serializer<T> deserializer;
     @Getter
     private final long startingOffset;
+    @Getter
     private final long endingOffset;
     private final EventSegmentReader input;
+    @Getter
+    private static SegmentInputStreamFactory factory;
```

Comment:
This should not be static. And it should be final.

Reply:
Taken care.